src/lib/frontend/d_cnf.ml

```diff
@@ -975,8 +975,10 @@ let rec mk_expr
                 E.mk_record [] ty
               | Tadt _ as ty ->
                 E.mk_constr (Uid.of_dolmen tcst) [] ty
-              | _ ->
-                assert false
+              | Tunit ->
+                E.void
+              | ty ->
+                Fmt.failwith "unexpected type %a@." Ty.print ty
```

Comment:
Is there a test that shows that Tunit is used? And that we crash if we don't match it here?

Reply:
There are 2000 regressions on `ae-format` without this case, so yes, we have tests :)
Actually the builtin `unit` type of `Dolmen` is an ADT but a distinct builtin type `Tunit` in Alt-Ergo. It's a corner corner corner case...

I tested this PR on `ae-format` and now there is no regression anymore ;)

Reply:
I will add a unit test to catch unit (sic)

Reply:
Finally, I removed the `Ty.Tunit` type of AE in #1148, so this corner case disappears.